src/coll_patterns/recursive_knomial.h

```diff
@@ -107,6 +107,18 @@ static inline int ucc_knomial_pattern_loop_done(ucc_knomial_pattern_t *p)
     return p->iteration == p->pow_radix_sup;
 }
 
+static inline int
+ucc_knomial_pattern_loop_first_iteration(ucc_knomial_pattern_t *p)
```

Comment:
change in allreduce_knomial.c and in reduce_scatter_knomial.c all the usages of "p->iteration == 0" and "p->iteration != 0" to calling this function instead and accordingly.